mypy/dmypy_server.py

```diff
@@ -232,12 +232,9 @@ def serve(self) -> None:
             # simplify the logic and always remove the file, since
             # that could cause us to remove a future server's
             # status file.)
+            server.close()
```

Comment:
Sully had a specific reason for the ordering of the closing of the socket and the unlinking of the status file; if you close the socket first, it's possible for a new client to start a new server before we get to remove the status file, and then we'd end up removing the wrong status file. See #5956.

Reply:
I think in this case it is just cleanliness. The bug fix was doing the file removal in `cmd_stop`, before any response is sent.